lightning/src/ln/channel.rs

```diff
@@ -2032,6 +2047,11 @@ impl<Signer: Sign> Channel<Signer> {
 			}
 		}
 
+		if inbound_dusted_htlc_msat + msg.amount_msat > self.get_max_balance_dust_htlc_msat() {
```

Comment:
I do not understand why we need to enforce this on the inbound leg. It should be sufficient to enforce it only on outbound HTLCs - ensuring *our* balance isn't dusted heavily. If our counterparty wants to burn all their balance to dust, that's on them.

Reply:
@ariard pointed out that this matters if we have to go claim the funds from this HTLC after it was forwarded onwards. The funds may be our counterparty's balance, but once we receive the preimage from the place we forwarded the HTLC to, the funds are now "ours" and may not be in the commitment transaction. Thus, this does need to be here.